lib/wallet/txdb.js

```diff
@@ -1777,17 +1777,16 @@ class TXDB {
       case types.REGISTER: {
         assert(i < tx.inputs.length);
 
-        const nameHash = covenant.getHash(0);
-        const prevout = tx.inputs[i].prevout;
-
-        const brv = await this.getReveal(nameHash, prevout);
-        assert(brv);
+        const coins = await this.getSpentCoins(tx);
```

Comment:
Is there any possibility that `coins` will be null here and throw an error on `coins[i]` ?

Reply:
`coins` should at least be an empty array.
https://github.com/handshake-org/hsd/blob/2d5ab277374ef751565caeaac84166c713ef9f4f/lib/wallet/txdb.js#L2760-L2777

Reply:
That said, I'm not sure if we've tested the `txdb.erase()` path. `walletdb.removeBlock()` exercises the `txdb.disconnect()` codepath. `txdb.unlockBalances()` is also called in `txdb.erase()`.

Reply:
We could test this using `txdb.abandon()`.

Reply:
Good looks, you are right that we haven't tested the `txdb.erase` path. Will add test coverage for that codepath.